### PR TITLE
Introduce CI pipeline

### PR DESCRIPTION
**Pull request by stafot**
#### Summary
WIP: Add test, introduce CI pipeline.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34456

#### Release Note

```release-note
Add CI pipeline. Automate releases.
```

### CODE REVIEW

File: scripts/release.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+release-notes --org mattermost --repo rotatorctl --start-sha $LATEST_RELEASE_NEXT_COMMIT_SHA --end-sha $FUTURE_RELEASE_SHA  --output ./build/_output/docs/relnote.md --required-author "" --branch main
+
+cat ./build/_output/docs/relnote.md | sed '/docs.k8s.io/ d' | sed -e "s/Release notes for/Release notes for ${CIRCLE_TAG}/g" | sed -e "s/Changelog since/Changelog since ${LATEST_RELEASE}/g"> ./build/_output/docs/relnote_parsed.md
+echo "\n" >> ./build/_output/docs/relnote_parsed.md
+echo "_Thanks to all our contributors!_" >> ./build/_output/docs/relnote_parsed.md
+mv ./build/_output/docs/relnote_parsed.md ./build/_output/docs/relnote.md
+
+make build && make build-mac
+mv  ./build/_output/bin/rotatorctl ./build/_output/bin/rotatorctl-linux
+hub release create -d -a ./build/_output/bin/rotatorctl-linux -a ./build/_output/bin/rotatorctl-darwin --file ./build/_output/docs/relnote.md ${CIRCLE_TAG}
```
Comment (stafot) [inline scripts/release.sh:18]: @spirosoik FYKI: `-d` creates draft. Eventually after some releases cutting testing we could remove it and publish the release not a draft of it.

File: .github/workflows/notify-release.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+jobs:
+  notify:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - shell: bash
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+        run: |
+          REPO=$(echo $GITHUB_CONTEXT | jq -r '.repository')
```
Comment (spirosoik) [inline .github/workflows/notify-release.yaml]: personal preference is to have everything in the makefile actually. not a blocker just mentioning

Comment (spirosoik) [reply .github/workflows/notify-release.yaml]: I mean it could be a `.sh` which is called by `Makefile`

**Review by spirosoik — APPROVED** [review]
minor comment

File: .github/workflows/notify-release.yaml (hunk 2 above)
Comment (stafot) [reply .github/workflows/notify-release.yaml]: Agree. I will refactor likewise. Thanks @spirosoik

**Review by stylianosrigas — APPROVED** [review]
LGTM

**Pull request merged**